### DISCUSSION TITLE
Declarations - new pattern proposal

### DISCUSSION THREAD

**Discussion by DeborahStevenson** [UKHomeOffice/design-system#378] [Ideas]
I have drafted a pattern for declarations and would like feedback from others across the Home Office. 

The pattern is intended to be a guide, images of best practice examples also need to be added.

PLEASE ADD ANY COMMENTS BY 10 MAY 2022.

### Declarations 

A declaration allows users to confirm that they have understood or agreed to something before they submit an online form.   

### Examples 
 
The following examples show how declaration are used in Home Office services. Use these examples as a guide only and adapt them for your service.  
 
You must ask a legal representative to sign off your declaration content.
  

INSERT IMAGE EXAMPLES

<img width="599" alt="Screenshot 2022-05-20 at 10 20 20" src="https://user-images.githubusercontent.com/90623478/169508630-ab2a4782-ce00-4662-aca2-de74ba462e8c.png">

<img width="473" alt="Screenshot 2022-05-20 at 10 35 44" src="https://user-images.githubusercontent.com/90623478/169508668-b9c880ae-d054-43a6-b64f-dc1df3797597.png">

 ### When to use this pattern 
   
Use this pattern when there are significant consequences if false information is given and the user needs to confirm that:   
  
- the information they’ve given is correct   
- they understand the consequences of not telling the truth   
- they understand the information they have been given  
- they need to agree to something  

  
### Writing for this pattern 

The declaration should be tailored to the service and include information about what will happen if the user makes a false declaration.  

Do:   

- make consequences clear using simple language 
- refer to the user as ‘you’ in the text, but ‘I’ in the action button   
- use ‘I confirm that’ or ‘I understand that’  
- link out to terms and conditions and privacy policy if needed   
- add a checkbox and ask the user to confirm that they understand and agree   
- get the content signed off by legal or policy

 
Do not: 

- duplicate terms and conditions or privacy policies 
- use the word ‘prosecute’ unless you are sure this is the case  
- use complex legal or policy terms if they can be explained simply 
- publish a declaration content that has not been sign off by legal or policy 

  
### When not to use this pattern  

  
Do not use this pattern to give users complex or complicated information about policy or legislation.  

If a bullet list is not needed, instead consider using ‘Accept and continue’ or ‘Agree and continue’ buttons along with a statement.  For example: 
 

By submitting this application, you are agreeing that the information you have provided is correct.  
 

### Error messages  
  
If your service uses this pattern add the error messages that you have used.  

  
### Services using this pattern   

This pattern has been used in the following services:  

- Electronic Visa Waiver  
- Register to Apply for US Global Entry  
- Registered Traveller  

 
### Research or evidence   

More research is needed on the pattern. If you have used this pattern add your research findings.  

### Accessibility   

Consider alternative ways a user can submit a declaration offline. 

If your service uses this pattern, let us know of any other insights you have on accessibility considerations. 
 
[Read ‘Error Prevention (Legal, Financial, Data)’ on the W3c website](https://www.w3.org/WAI/WCAG21/Understanding/error-prevention-legal-financial-data.html) (Opens new tab)

**Comment by willrichards** [comment]
@DeborahStevenson The default checkbox pattern is inherited from GOV.UK here: https://design-system.service.gov.uk/components/checkboxes/ and renders like this in our design system:

![Screenshot 2022-05-06 at 11 16 51](https://user-images.githubusercontent.com/606378/167113439-55bf8180-90a5-4c10-89f8-c9b8d481fcf2.png)

Can we look at using this pattern for the image examples?

**Comment by guermanc** [comment]
There are a number of potential things to look out surrounding basic component accessibility, such as correct uses of headings and lists, correctly implemented and labeled checkboxes, and the use of a descriptive error message if a user doesn't select the checkbox.

To better meet https://www.w3.org/WAI/WCAG21/Understanding/error-prevention-legal-financial-data.html I would reword the bullet point around the use of a checkbox, and make the use of one mandatory.

Due to the potential complexity of the language and the severity of consequences to the user, there are numerous considerations to be made regarding making the content usable for people with cognitive and learning disabilities, as well non-native English speakers. These should be clearly signposted as part of the pattern.

We cover readability basics on https://design.homeoffice.gov.uk/accessibility/readability. There are also numerous objectives that go into more depth in the Making Content Usable for People with Cognitive and Learning Disabilities W3C Working Group Note, the two that are most relevant to this pattern are as follows, but other guidance here is also likely to be relevant.

Objective 3: Use Clear and Understandable Content
- User Story - https://www.w3.org/TR/coga-usable/#objective-3-use-clear-and-understandable-content
- Design Guide - https://www.w3.org/TR/coga-usable/#objective-3-use-clear-and-understandable-content-0

Objective 7: Provide Help and Support 
- User Story - https://www.w3.org/TR/coga-usable/#objective-7-provide-help-and-support
- Design Guide - https://www.w3.org/TR/coga-usable/#objective-7-provide-help-and-support-0

It's worth thinking about what alternative routes and workarounds a user might have if they do not wish to submit the declaration online. There must be other ways for the user to complete the journey and additional user research should be done to support the use of a declaration screen.

**Comment by DeborahStevenson** [comment]
I have updated the proposal to include the comments made.

**Reply by naomiwilcock** [reply]
@DeborahStevenson sorry for the lateness of this but we saw some accessibility issues with using a checkbox for declarations in some immigration and visas services as there is no direct heading relating to the checkbox. I suggest we look at the EU Settlement Scheme/visa and immigration application declaration pattern which was tested for a way to do a declaration pattern without a checkbox. The checkbox also appears to duplicate the action the user takes when they select the primary button.
